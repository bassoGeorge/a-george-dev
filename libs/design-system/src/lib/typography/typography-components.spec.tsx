import { render } from '@testing-library/react';
import {
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  Heading5,
  Heading6,
  BodyXl,
  BodyLg,
  BodyMd,
  Body,
  BodySm,
  BodyXs,
  Interface2Xl,
  InterfaceXl,
  InterfaceLg,
  InterfaceMd,
  Interface,
  InterfaceSm,
} from './typography-components';

describe('Typography', () => {
  it('should add add correct styling for heading types', async () => {
    const { baseElement } = await render(
      <div>
        <Heading1>Heading 1</Heading1>
        <Heading2>Heading 2</Heading2>
        <Heading3>Heading 3</Heading3>
        <Heading4>Heading 4</Heading4>
        <Heading5>Heading 5</Heading5>
        <Heading6>Heading 6</Heading6>
      </div>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should add add correct styling for body types', async () => {
    const { baseElement } = await render(
      <div>
        <BodyXl>Body Xl</BodyXl>
        <BodyLg>Body Lg</BodyLg>
        <BodyMd>Body Md</BodyMd>
        <Body>Body</Body>
        <BodySm>Body Sm</BodySm>
        <BodyXs>Body Xs</BodyXs>
      </div>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should add add correct styling for interface types', async () => {
    const { baseElement } = await render(
      <div>
        <Interface2Xl>Interface 2Xl</Interface2Xl>
        <InterfaceXl>Interface Xl</InterfaceXl>
        <InterfaceLg>Interface Lg</InterfaceLg>
        <InterfaceMd>Interface Md</InterfaceMd>
        <Interface>Interface</Interface>
        <InterfaceSm>Interface Sm</InterfaceSm>
      </div>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
