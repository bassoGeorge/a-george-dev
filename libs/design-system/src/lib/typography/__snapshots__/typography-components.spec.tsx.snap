// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should add add correct styling for body types 1`] = `
<body>
  <div>
    <div>
      <p
        class="font-body text-xl"
      >
        Body Xl
      </p>
      <p
        class="font-body text-lg"
      >
        Body Lg
      </p>
      <p
        class="font-body text-md"
      >
        Body Md
      </p>
      <p
        class="font-body text-base"
      >
        Body
      </p>
      <p
        class="font-body text-sm"
      >
        Body Sm
      </p>
      <p
        class="font-body text-xs"
      >
        Body Xs
      </p>
    </div>
  </div>
</body>
`;

exports[`Typography should add add correct styling for heading types 1`] = `
<body>
  <div>
    <div>
      <h1
        class="font-heading text-4xl tablet:text-5xl desktop:text-6xl"
      >
        Heading 1
      </h1>
      <h2
        class="font-heading text-3xl tablet:text-4xl desktop:text-5xl"
      >
        Heading 2
      </h2>
      <h3
        class="font-heading text-2xl tablet:text-3xl desktop:text-4xl"
      >
        Heading 3
      </h3>
      <h4
        class="font-heading text-xl tablet:text-2xl desktop:text-3xl"
      >
        Heading 4
      </h4>
      <h5
        class="font-heading text-lg tablet:text-xl desktop:text-2xl"
      >
        Heading 5
      </h5>
      <h6
        class="font-heading text-md tablet:text-lg desktop:text-xl"
      >
        Heading 6
      </h6>
    </div>
  </div>
</body>
`;

exports[`Typography should add add correct styling for interface types 1`] = `
<body>
  <div>
    <div>
      <span
        class="font-interface text-2xl leading-none"
      >
        Interface 2Xl
      </span>
      <span
        class="font-interface text-xl leading-none"
      >
        Interface Xl
      </span>
      <span
        class="font-interface text-lg leading-none"
      >
        Interface Lg
      </span>
      <span
        class="font-interface text-md leading-none"
      >
        Interface Md
      </span>
      <span
        class="font-interface leading-none"
      >
        Interface
      </span>
      <span
        class="font-interface text-sm leading-none"
      >
        Interface Sm
      </span>
    </div>
  </div>
</body>
`;
